projects/components/src/utils/id-generator/id-generator.ts

```diff
@@ -0,0 +1,26 @@
+/*!
+ * Copyright 2019 VMware, Inc.
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+/**
+ * Object containing the logic to generate unique identifiers whenever it's {@link IdGenerator#id} method is called
+ */
+export class IdGenerator {
+    /**
+     * Variable that is incremented whenever a new ID is needed
+     */
+    private static idCounter = 0;
+
+    /**
+     * Getter that returns the a unique ID
```

Comment:
No need to say getter, the code already says that. Having said that, a getter that always returns a different value seems wrong. Please rename to `generate` following the pattern from https://mongodb.github.io/mongo-java-driver/3.8/javadoc/org/bson/codecs/IdGenerator.html